src/bd/amazed/pdfscissors/model/TaskPdfOpen.java

```diff
@@ -54,27 +50,19 @@ public TaskPdfOpen(File file, int groupType, boolean shouldCreateStackView, Comp
 
 			if (shouldCreateStackView && groupType != PageGroup.GROUP_TYPE_INDIVIDUAL) {
 				setProgress(0);
-				PropertyChangeListener propertyChangeListener = new PropertyChangeListener() {
-					@Override
-					public void propertyChange(PropertyChangeEvent evt) {
-						firePropertyChange(evt.getPropertyName(), evt.getOldValue(), evt.getNewValue());
-					}
-				};
-
-				for (int i = 0; i < pageGroups.size(); i++) {
-					PageGroup pageGroup = pageGroups.elementAt(i);
-					BufferedImage image = cropper.getImage(propertyChangeListener, pageGroup);
-
-					if (image == null) {
-						debug("Ups.. null image for " + pdfFile.getNormalizedFile());
-					} else {
-						debug("PDF loaded " + pageGroup + " from " + pdfFile.getNormalizedFile());
-					}
-					pageGroup.setStackImage(image);
-
-				}
+				TaskCreateThumbnails thumbnailTask = launchThumbnailTask(pdfFile, pageGroups);
+				
+				//make this thread wait, until the thumbnails-thread has finished:
+				// (images are previews for stacked-pages, i.e. required)
+				thumbnailTask.get();
+				
 				setProgress(100);
 			}
+			else {
+				//start thumbnails-thread without waiting for it to finish
+				// (images are only previews for single pages, i.e. not required)
+				launchThumbnailTask(pdfFile, pageGroups);

```

Comment:
It has been such a long time ago, my memory isnot helping much :)
I just vaguely remember it was something todo with memory.

Anyways, I tested with 100MB scanned pdf, there was no problem. So I will merge this pull request now. :+1: 

Threaded thumbnail generation gave me an idea about UI though. Perhaps user does not need to wait for stacked-view by default, perhaps the progress bar could be in the thumbnail view itself, for example, below the stacked-view-thumbnail with a (X) cancel button. Looks like I have something to play with in coming weekends :)